R/plotSpectra-PTM.R

```diff
@@ -0,0 +1,482 @@
+#'
+#' @importFrom graphics axis plot.new plot.window plot.xy strwidth
+#'
+#' @importFrom graphics text title
+#'
+#' @importFrom grDevices xy.coords
+#' 
+#' @noRd
+.plot_single_spectrum_PTM <- function(x, xlab = "m/z", ylab = "intensity",
+                                      xlim = numeric(),
+                                      ylim = numeric(), main = character(),
+                                      col = c(y = "darkred",
+                                              b = "darkblue",
+                                              acxy = "darkgreen", 
+                                              other = "grey40"),
+                                      labels = list(),
+                                      labelCol = col, labelCex = 1, labelSrt = 0,
+                                      labelAdj = NULL, labelPos = 3,
+                                      labelOffset = 0.5, minorTicks = TRUE,
+                                      ppm = 20, deltaMzData = NULL) {
+    v <- peaksData(x)[[1L]]
+    mzs <- v[, "mz"]
+    ints <- v[, "intensity"]
+    old_par <- par(no.readonly = TRUE)
+    
+    if (!length(xlim))
+        suppressWarnings(xlim <- range(mzs, na.rm = TRUE))
+    if (!length(ylim))
+        suppressWarnings(
+            ylim <- range(c(0, max(abs(ints), na.rm = TRUE))))
+    if (any(is.infinite(xlim)))
+        xlim <- c(0, 0)
+    if (any(is.infinite(ylim)))
+        ylim <- c(0, 0)
+    if (length(main)) {
+        par(mar = c(4,4,1.4,2))
+    } else {par(mar = c(4,4,0.5,2))}
+    plot.new()
+    plot.window(xlim = xlim, ylim = ylim)
+    
+    peptide_sequence <- names(labels)
+    labels <- labels[[1]]
+    wdths <- max(strwidth(labels, cex = labelCex)) / 2
+    usr_lim <- par("usr")
+    ylim[2L] <- ylim[2L] + 0.4*diff(ylim)
+    xlim[1L] <- xlim[1L] - wdths
+    xlim[2L] <- xlim[2L] + wdths
+    plot.window(xlim = xlim, ylim = ylim)
+    
+    peakCol <- ifelse(grepl("b", labels), col[["b"]],
+                      ifelse(grepl("y", labels), col[["y"]], 
+                             ifelse(grepl("a|c|x|z", labels), col[["acxy"]], 
+                                    col[["other"]])))
+    
+    labelCol <- ifelse(grepl("b", labels), col[["b"]],
+                       ifelse(grepl("y", labels), col[["y"]], col[["acxy"]]))
+    
+    text(mzs, ints, labels = labels, adj = labelAdj, pos = labelPos,
+         col = labelCol, cex = labelCex, srt = labelSrt, offset = labelOffset)
+    
+    .draw_psmanno(x, mzs, ints, col, labels, peptide_sequence)
+    
+    plot.xy(xy.coords(mzs, ints), type = "h", col = peakCol)
+    
+    major_ticks <- pretty(mzs, n = 8)
+    axis(side = 1, lwd = 1, at = major_ticks, pos = c(0,0),
+         col.ticks = "grey45", col = "grey45")
+    
+    if (minorTicks) {
+        minor_ticks <- unlist(lapply(seq_along(major_ticks[-length(major_ticks)]),
+                                     function(i) {seq(major_ticks[i],
```

Comment:
Always start the body of the function on a new line. If it's a one-line body, not `{}`